diff -urN gcc-4.6.0-orig//gcc/gcc.c gcc-4.6.0/gcc/gcc.c
--- gcc-4.6.0-orig//gcc/gcc.c	2011-05-01 14:48:36.835034414 -0400
+++ gcc-4.6.0/gcc/gcc.c	2011-05-01 14:49:30.730034372 -0400
@@ -6418,7 +6418,7 @@
    * each spec listed, the string is overwritten at token boundaries
    * (':') with '\0', an effect of strtok_r().
    */
-  GET_ENVIRONMENT (specs_file, "GCC_SPECS");
+  specs_file = getenv ("GCC_SPECS");
   if (specs_file && (strlen(specs_file) > 0))
     {
       char *spec, *saveptr;
