--- poppler-python/poppler.defs	2010-07-14 21:36:22.166804875 +0200
+++ poppler-new/poppler.defs	2010-07-14 21:16:55.460805922 +0200
@@ -915,6 +915,7 @@
 
 (define-function document_new_from_file
   (c-name "poppler_document_new_from_file")
+  (caller-owns-return #t)
   (return-type "PopplerDocument*")
   (parameters
     '("const-char*" "uri")
@@ -925,6 +926,7 @@
 
 (define-function document_new_from_data
   (c-name "poppler_document_new_from_data")
+  (caller-owns-return #t)
   (return-type "PopplerDocument*")
   (parameters
     '("char*" "data")
@@ -963,6 +965,7 @@
 (define-method get_page
   (of-object "PopplerDocument")
   (c-name "poppler_document_get_page")
+  (caller-owns-return #t)
   (return-type "PopplerPage*")
   (parameters
     '("int" "index")
@@ -972,6 +975,7 @@
 (define-method get_page_by_label
   (of-object "PopplerDocument")
   (c-name "poppler_document_get_page_by_label")
+  (caller-owns-return #t)
   (return-type "PopplerPage*")
   (parameters
     '("const-char*" "label")
@@ -993,6 +997,7 @@
 (define-method find_dest
   (of-object "PopplerDocument")
   (c-name "poppler_document_find_dest")
+  (caller-owns-return #t)
   (return-type "PopplerDest*")
   (parameters
     '("const-gchar*" "link_name")
@@ -1002,6 +1007,7 @@
 (define-method get_form_field
   (of-object "PopplerDocument")
   (c-name "poppler_document_get_form_field")
+  (caller-owns-return #t)
   (return-type "PopplerFormField*")
   (parameters
     '("gint" "id")
@@ -1025,6 +1031,7 @@
 (define-method copy
   (of-object "PopplerIndexIter")
   (c-name "poppler_index_iter_copy")
+  (caller-owns-return #t)
   (return-type "PopplerIndexIter*")
 )
 
@@ -1049,6 +1056,7 @@
 (define-method get_action
   (of-object "PopplerIndexIter")
   (c-name "poppler_index_iter_get_action")
+  (caller-owns-return #t)
   (return-type "PopplerAction*")
 )
 
