--- poppler.defs.old	2009-09-27 11:07:59.000000000 +0200
+++ poppler.defs	2011-09-29 02:53:25.000000000 +0200
@@ -1676,30 +1676,6 @@
   )
 )
 
-(define-method get_thumbnail_pixbuf
-  (of-object "PopplerPage")
-  (c-name "poppler_page_get_thumbnail_pixbuf")
-  (return-type "GdkPixbuf*")
-  (unblock-threads #t)
-)
-
-(define-method render_selection_to_pixbuf
-  (of-object "PopplerPage")
-  (c-name "poppler_page_render_selection_to_pixbuf")
-  (return-type "none")
-  (unblock-threads #t)
-  (parameters
-    '("gdouble" "scale")
-    '("int" "rotation")
-    '("GdkPixbuf*" "pixbuf")
-    '("PopplerRectangle*" "selection")
-    '("PopplerRectangle*" "old_selection")
-    '("PopplerSelectionStyle" "style")
-    '("GdkColor*" "glyph_color")
-    '("GdkColor*" "background_color")
-  )
-)
-
 (define-method render
   (of-object "PopplerPage")
   (c-name "poppler_page_render")
